plugins/out_es/es_conf.c

```diff
@@ -153,8 +157,37 @@ struct flb_elasticsearch *flb_es_conf_create(struct flb_output_instance *ins,
             flb_es_conf_destroy(ctx);
             return NULL;
         }
+        flb_plg_debug(ctx->ins, "extracted cloud_host: '%s'", cloud_host);
+
+        flb_plg_debug(ctx->ins,
+                      "check whether extracted cloud_host '%s' contains a port",
+                      cloud_host);
+        cloud_host = strtok_r(cloud_host, ":", &saveptr);
+        flb_plg_debug(ctx->ins, "cloud_host without port: '%s'", cloud_host);
+        if (cloud_host == NULL) {
+            flb_plg_error(ctx->ins, "cannot extract cloud_host");
+            flb_es_conf_destroy(ctx);
+            return NULL;
+        }
+
+        cloud_port_char = strtok_r(NULL, ":", &saveptr);
+        flb_plg_debug(ctx->ins, "extracted cloud_port_char: '%s'", cloud_port_char);
+
+        if (cloud_port_char != NULL) {
+            cloud_host_port = (int)strtol(cloud_port_char, (char **)NULL, 10);
+            flb_plg_debug(ctx->ins, "converted cloud_port_char to port int: '%i'", cloud_host_port);
+        }
+
+        if (cloud_host_port == 0) {
+            cloud_host_port = cloud_port;
+        }
+        flb_plg_debug(ctx->ins,
+                      "checked whether extracted port was null and set it to "
+                      "default https port or not. Outcome: '%i'",
+                      cloud_host_port);
+
         ins->host.name = cloud_host;
```

Comment:
If you can, I request one more thing.

If user set`Host` and `Cloud_ID`, an original `ins->host.name` will not be freed.
Could you check `ins->host.name` is NULL or not ?
If it is not NULL, it should be freed using `flb_sds_destroy` before saving `cloud_host`.

```
[SERVICE]
    Log_Level debug

[INPUT]
    Name dummy
[OUTPUT]
    Name es
    Host google.com
    Cloud_ID custom-port:dXMtY2VudHJhbDEuZ2NwLmNsb3VkLmVzLmlvOjkyNDMkYWMzMWViYjkwMjQxNzczMTU3MDQzYzM0ZmQyNmZkNDYkYTRjMDYyMzBlNDhjOGZjZTdiZTg4YTA3NGEzYmIzZTA=
```

```
$ valgrind --leak-check=full bin/fluent-bit -c a.conf 
==90091== Memcheck, a memory error detector
==90091== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==90091== Using Valgrind-3.15.0 and LibVEX; rerun with -h for copyright info
==90091== Command: bin/fluent-bit -c a.conf
==90091== 
Fluent Bit v2.0.0
* Copyright (C) 2015-2022 The Fluent Bit Authors
* Fluent Bit is a CNCF sub-project under the umbrella of Fluentd
* https://fluentbit.io

[2022/07/31 14:18:51] [ info] Configuration:
[2022/07/31 14:18:51] [ info]  flush time     | 1.000000 seconds
[2022/07/31 14:18:51] [ info]  grace          | 5 seconds
[2022/07/31 14:18:51] [ info]  daemon         | 0
[2022/07/31 14:18:51] [ info] ___________
[2022/07/31 14:18:51] [ info]  inputs:
[2022/07/31 14:18:51] [ info]      dummy
[2022/07/31 14:18:51] [ info] ___________
[2022/07/31 14:18:51] [ info]  filters:
[2022/07/31 14:18:51] [ info] ___________
[2022/07/31 14:18:51] [ info]  outputs:
[2022/07/31 14:18:51] [ info]      es.0
[2022/07/31 14:18:51] [ info] ___________
[2022/07/31 14:18:51] [ info]  collectors:
[2022/07/31 14:18:51] [ info] [fluent bit] version=2.0.0, commit=f023eb8d93, pid=90091
[2022/07/31 14:18:51] [debug] [engine] coroutine stack size: 24576 bytes (24.0K)
[2022/07/31 14:18:51] [ info] [storage] version=1.2.0, type=memory-only, sync=normal, checksum=disabled, max_chunks_up=128
[2022/07/31 14:18:51] [ info] [cmetrics] version=0.3.5
[2022/07/31 14:18:52] [debug] [dummy:dummy.0] created event channels: read=21 write=22
[2022/07/31 14:18:52] [debug] [es:es.0] created event channels: read=23 write=24
[2022/07/31 14:18:52] [debug] [output:es:es.0] extracted cloud_host: 'ac31ebb90241773157043c34fd26fd46.us-central1.gcp.cloud.es.io:9243'
[2022/07/31 14:18:52] [debug] [output:es:es.0] check whether extracted cloud_host 'ac31ebb90241773157043c34fd26fd46.us-central1.gcp.cloud.es.io:9243' contains a port
[2022/07/31 14:18:52] [debug] [output:es:es.0] cloud_host without port: 'ac31ebb90241773157043c34fd26fd46.us-central1.gcp.cloud.es.io'
[2022/07/31 14:18:52] [debug] [output:es:es.0] extracted cloud_port_char: '9243'
[2022/07/31 14:18:52] [debug] [output:es:es.0] converted cloud_port_char to port int: '9243'
[2022/07/31 14:18:52] [debug] [output:es:es.0] checked whether extracted port was null and set it to default https port or not. Outcome: '9243'
[2022/07/31 14:18:52] [debug] [output:es:es.0] host=ac31ebb90241773157043c34fd26fd46.us-central1.gcp.cloud.es.io port=9243 uri=/_bulk index=fluent-bit type=_doc
[2022/07/31 14:18:52] [ info] [output:es:es.0] worker #0 started
[2022/07/31 14:18:52] [debug] [router] default match rule dummy.0:es.0
[2022/07/31 14:18:52] [ info] [sp] stream processor started
[2022/07/31 14:18:52] [ info] [output:es:es.0] worker #1 started
[2022/07/31 14:18:52] [debug] [input chunk] update output instances with new chunk size diff=26
[2022/07/31 14:18:53] [debug] [task] created task=0x4ee4e00 id=0 OK
[2022/07/31 14:18:53] [debug] [output:es:es.0] task_id=0 assigned to thread #0
[2022/07/31 14:18:53] [debug] [input chunk] update output instances with new chunk size diff=26
^C[2022/07/31 14:18:53] [engine] caught signal (SIGINT)
[2022/07/31 14:18:53] [debug] [task] created task=0x4f50b50 id=1 OK
[2022/07/31 14:18:53] [debug] [output:es:es.0] task_id=1 assigned to thread #1
[2022/07/31 14:18:53] [ warn] [engine] service will shutdown in max 5 seconds
[2022/07/31 14:18:53] [ info] [input] pausing dummy.0
[2022/07/31 14:18:53] [debug] [http_client] not using http_proxy for header
[2022/07/31 14:18:53] [debug] [http_client] not using http_proxy for header
[2022/07/31 14:18:53] [error] [http_client] broken connection to ac31ebb90241773157043c34fd26fd46.us-central1.gcp.cloud.es.io:9243 ?
[2022/07/31 14:18:53] [ warn] [output:es:es.0] http_do=-1 URI=/_bulk
[2022/07/31 14:18:53] [debug] [upstream] KA connection #48 to ac31ebb90241773157043c34fd26fd46.us-central1.gcp.cloud.es.io:9243 is now available
[2022/07/31 14:18:53] [debug] [upstream] KA connection #48 to ac31ebb90241773157043c34fd26fd46.us-central1.gcp.cloud.es.io:9243 has been disconnected by the remote service
[2022/07/31 14:18:53] [debug] [out flush] cb_destroy coro_id=0
[2022/07/31 14:18:54] [ info] [task] dummy/dummy.0 has 2 pending task(s):
[2022/07/31 14:18:54] [ info] [task]   task_id=0 still running on route(s): es/es.0 
[2022/07/31 14:18:54] [ info] [task]   task_id=1 still running on route(s): es/es.0 
[2022/07/31 14:18:54] [ info] [input] pausing dummy.0
[2022/07/31 14:18:54] [debug] [output:es:es.0] HTTP Status=400 URI=/_bulk
[2022/07/31 14:18:54] [error] [output:es:es.0] HTTP status=400 URI=/_bulk, response:
Client sent an HTTP request to an HTTPS server.


[2022/07/31 14:18:54] [debug] [upstream] KA connection #49 to ac31ebb90241773157043c34fd26fd46.us-central1.gcp.cloud.es.io:9243 is now available
[2022/07/31 14:18:54] [debug] [upstream] KA connection #49 to ac31ebb90241773157043c34fd26fd46.us-central1.gcp.cloud.es.io:9243 has been disconnected by the remote service
[2022/07/31 14:18:54] [debug] [out flush] cb_destroy coro_id=0
[2022/07/31 14:18:55] [ info] [input] pausing dummy.0
[2022/07/31 14:18:56] [ info] [input] pausing dummy.0
[2022/07/31 14:18:57] [ info] [input] pausing dummy.0
[2022/07/31 14:18:58] [ info] [task] dummy/dummy.0 has 2 pending task(s):
[2022/07/31 14:18:58] [ info] [task]   task_id=0 still running on route(s): es/es.0 
[2022/07/31 14:18:58] [ info] [task]   task_id=1 still running on route(s): es/es.0 
[2022/07/31 14:18:58] [ info] [engine] service has stopped (2 pending tasks)
[2022/07/31 14:18:58] [ info] [input] pausing dummy.0
[2022/07/31 14:18:58] [debug] [task] destroy task=0x4ee4e00 (task_id=0)
[2022/07/31 14:18:58] [debug] [task] destroy task=0x4f50b50 (task_id=1)
[2022/07/31 14:18:58] [ info] [output:es:es.0] thread worker #0 stopping...
[2022/07/31 14:18:58] [ info] [output:es:es.0] thread worker #0 stopped
[2022/07/31 14:18:58] [ info] [output:es:es.0] thread worker #1 stopping...
[2022/07/31 14:18:58] [ info] [output:es:es.0] thread worker #1 stopped
==90091== 
==90091== HEAP SUMMARY:
==90091==     in use at exit: 27 bytes in 1 blocks
==90091==   total heap usage: 1,457 allocs, 1,456 frees, 1,272,108 bytes allocated
==90091== 
==90091== 27 bytes in 1 blocks are definitely lost in loss record 1 of 1
==90091==    at 0x483B7F3: malloc (in /usr/lib/x86_64-linux-gnu/valgrind/vgpreload_memcheck-amd64-linux.so)
==90091==    by 0x18F6FF: flb_malloc (flb_mem.h:79)
==90091==    by 0x18F8E2: sds_alloc (flb_sds.c:41)
==90091==    by 0x18FA2A: flb_sds_create_size (flb_sds.c:93)
==90091==    by 0x1887D8: flb_env_var_translate (flb_env.c:180)
==90091==    by 0x19B6AF: flb_output_set_property (flb_output.c:573)
==90091==    by 0x177A6B: service_configure_plugin (fluent-bit.c:747)
==90091==    by 0x177EAD: service_configure (fluent-bit.c:854)
==90091==    by 0x1786D9: flb_main (fluent-bit.c:1142)
==90091==    by 0x178878: main (fluent-bit.c:1217)
==90091== 
==90091== LEAK SUMMARY:
==90091==    definitely lost: 27 bytes in 1 blocks
==90091==    indirectly lost: 0 bytes in 0 blocks
==90091==      possibly lost: 0 bytes in 0 blocks
==90091==    still reachable: 0 bytes in 0 blocks
==90091==         suppressed: 0 bytes in 0 blocks
==90091== 
==90091== For lists of detected and suppressed errors, rerun with: -s
==90091== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
```

Reply:
@nokute78 Processed the feedback:

```
==36416== 
==36416== HEAP SUMMARY:
==36416==     in use at exit: 106,442 bytes in 3,650 blocks
==36416==   total heap usage: 16,668 allocs, 13,018 frees, 35,228,114 bytes allocated
==36416== 
==36416== LEAK SUMMARY:
==36416==    definitely lost: 0 bytes in 0 blocks
==36416==    indirectly lost: 0 bytes in 0 blocks
==36416==      possibly lost: 0 bytes in 0 blocks
==36416==    still reachable: 106,442 bytes in 3,650 blocks
==36416==         suppressed: 0 bytes in 0 blocks
==36416== Reachable blocks (those to which a pointer was found) are not shown.
==36416== To see them, rerun with: --leak-check=full --show-leak-kinds=all
==36416== 
==36416== Use --track-origins=yes to see where uninitialised values come from
==36416== For lists of detected and suppressed errors, rerun with: -s
==36416== ERROR SUMMARY: 24739 errors from 1000 contexts (suppressed: 0 from 0)
```